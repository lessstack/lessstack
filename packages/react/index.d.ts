// import defaultExport from "./build";
export * from "./build";
// export default defaultExport;
