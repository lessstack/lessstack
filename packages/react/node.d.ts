// import defaultExport from "./build";
export * from "./build/node";
// export default defaultExport;
