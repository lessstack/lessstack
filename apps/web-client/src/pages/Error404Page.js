import React from "react";
import { Link } from "react-router-dom";
import { ResponsePropType } from "@hmr/config-webpack/utils";

const HelloWorldPage = ({ response }) => {
  if (response) {
    response.statusCode = 404;
  }

  return (
    <div>
      <h1>404 Not Found</h1>
      <p>
        <Link to="/">Back to home</Link>
      </p>
    </div>
  );
};

HelloWorldPage.defaultProps = {
  response: null,
};

HelloWorldPage.propTypes = {
  response: ResponsePropType,
};

export default HelloWorldPage;
