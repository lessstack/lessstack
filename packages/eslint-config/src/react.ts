import type { ESLint } from "eslint";

const reactConfig: ESLint.ConfigData = {
  extends: [
    "plugin:react/recommended",
    "plugin:react/jsx-runtime",
    "plugin:react-hooks/recommended",
    "plugin:jsx-a11y/recommended",
  ],
  parserOptions: {
    ecmaFeatures: {
      jsx: true,
    },
  },
  plugins: ["react", "react-hooks"],
  rules: {
    /**
     * Boolean properties must starts with "is" or "has" prefix.
     */
    "react/boolean-prop-naming": [
      "error",
      {
        rule: "^(is|has)[A-Z]([A-Za-z0-9]?)+",
        validateNested: false,
      },
    ],
    /**
     * This rule is impracticable as it prohibits dynamic
     * assignment even with enforced type or proptype.
     * @see: https://github.com/jsx-eslint/eslint-plugin-react/issues/1555
     */
    "react/button-has-type": "off",
    /**
     * Use destructuring assignment instead of defaultProps they might
     * be deprecated in the future.
     * @see: https://twitter.com/dan_abramov/status/1133878326358171650
     */
    "react/default-props-match-prop-types": ["error"],
    /**
     * Destructuring props enforce the use of a single default value by
     * destructuring assignment.
     * @todo: find rule for enforcing destructuring assignment of optional
     * properties.
     */
    "react/destructuring-assignment": [
      "error",
      "always",
      { destructureInSignature: "always" },
    ],
    "react/function-component-definition": [
      "error",
      {
        namedComponents: ["arrow-function"],
      },
    ],
    "react/require-default-props": [
      "error",
      {
        functions: "defaultArguments",
      },
    ],
    "react-hooks/exhaustive-deps": "error",
    "react-hooks/rules-of-hooks": "error",
  },
  settings: {
    react: {
      version: "detect",
    },
  },
};

export = reactConfig;
