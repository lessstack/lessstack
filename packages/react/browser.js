/* eslint-env node */
module.exports = require("./build/browser");
