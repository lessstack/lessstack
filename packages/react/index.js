/* eslint-env node */
module.exports = require("./build");
