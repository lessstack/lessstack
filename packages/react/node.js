/* eslint-env node */
module.exports = require("./build/node");
