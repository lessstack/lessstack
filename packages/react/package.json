{
  "name": "@lessstack/react",
  "version": "0.1.6",
  "main": "build/index.js",
  "license": "MIT",
  "author": "Cahnory (https://github.com/cahnory)",
  "repository": "https://github.com/lessstack/lessstack",
  "scripts": {
    "lint": "eslint",
    "build": "tsc --outDir build",
    "dev": "tsc --outDir build --watch"
  },
  "dependencies": {
    "@loadable/component": "^5.15.2",
    "@loadable/server": "^5.15.2"
  },
  "devDependencies": {
    "@lessstack/eslint-config": "workspace:*",
    "@lessstack/prettier-config": "workspace:*",
    "@lessstack/typescript-config": "workspace:*",
    "@types/loadable__component": "^5.13.4",
    "@types/loadable__server": "^5.12.6",
    "@types/node": "^18.8.5",
    "@types/react": "^18.0.21",
    "@types/react-dom": "^18.0.6",
    "eslint": "^8.2.0",
    "prettier": "^2.4.1",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "typescript": "^4.7.4"
  },
  "peerDependencies": {
    "react": "^18.2.0",
    "react-dom": "^18.2.0"
  }
}
