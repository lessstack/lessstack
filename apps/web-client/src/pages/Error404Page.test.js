/**
 * @jest-environment jsdom
 */
import React from "react";
import { render } from "@testing-library/react";
import Error404Page from "./Error404Page";
import JestContext from "../jest";

it("stets response.statusCode", () => {
  const response = { headers: {} };
  render(
    <JestContext>
      <Error404Page response={response} />
    </JestContext>,
  );

  expect(response.statusCode).toEqual(404);
});
