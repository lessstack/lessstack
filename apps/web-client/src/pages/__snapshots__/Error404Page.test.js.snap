// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div>
  <div>
    <h1>
      404 Not Found
    </h1>
    <p>
      <a
        href="/"
      >
        Back to home
      </a>
    </p>
  </div>
</div>
`;
